src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -533,6 +533,104 @@ function transformTopLevelAtom(atom, field) {
   }
 }
 
+function relativeTimeToDate(text, now = new Date()) {
+  let parts = text.split(' ');
+  if (!parts.length) {
+    return { status: 'error', info: 'Not a time string' };
+  }
+
+  // Filter out whitespace
+  parts = parts.filter((part) => part !== '');
+
+  const future = parts[0] === 'in';
+  const past = parts[parts.length - 1] === 'ago';
+
+  if (!future && !past) {
+    return { status: 'error', info: "Time should either start with 'in' or end with 'ago'" };
+  }
+
+  if (future && past) {
+    return {
+      status: 'error',
+      info: "Time cannot have both 'in' and 'ago'",
+    };
+  }
+
+  // strip the 'ago' or 'in'
+  if (future) {
+    parts = parts.slice(1);
+  } else if (past) {
+    parts = parts.slice(0, parts.length - 1);
+  }
+
+  if (parts.length % 2 !== 0) {
+    return {
+      status: 'error',
+      info: 'Invalid time string. Dangling unit or number.',
+    };
+  }
+
+  const pairs = [];
+  while(parts.length) {
+    pairs.push([ parts.shift(), parts.shift() ]);
+  }
+
+  let seconds = 0;
+  for (const [num, interval] of pairs) {
+    const val = Number(num);
+    if (!Number.isInteger(val)) {
```

Comment:
With only whole integer values allowed perhaps we should add a test that expects to fail when someone tries something like `2.5 days`.

Reply:
From the code, as interpolated to seconds, it seems that it should work with non Integer values no?

Reply:
Makes sense it would work, but we would want to test for it if we decide to allow it (or disallow it as currently).

Reply:
let's stick with integers for now.